packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmDictionaryTests.kt

```diff
@@ -606,6 +690,429 @@ internal abstract class ManagedDictionaryTester<T>(
+                assertFailsWith<RealmException> {
+                    entries.remove(realmDictionaryEntryOf(dataSet[1].first, dataSet[1].second))
+                }
+            }
+        }
+    }
+
+    override fun entries_removeAll() {
+        // Ignore ByteArray and RealmObject: structural equality cannot be assessed for these types
+        if (classifier != ByteArray::class && classifier != RealmObject::class) {
+            val dataSet = typeSafetyManager.dataSetToLoad
+
+            errorCatcher {
+                realm.writeBlocking {
+                    val dictionary = typeSafetyManager.createContainerAndGetCollection(this)
+                    dictionary.putAll(dataSet)
+                    val entries = dictionary.entries
+
+                    // This list of entries contains an entry whose value doesn't match what is in
+                    // the dictionary
+                    val bogusEntriesToRemove = listOf(
+                        realmDictionaryEntryOf(dataSet[0].first, dataSet[1].second)
+                    )
+                    assertFalse(entries.removeAll(bogusEntriesToRemove))
+
+                    // This list of entries contains an entry that is present in the dictionary and
+                    // another one that isn't and it returns true anyway since something got removed
+                    val entriesToRemove = listOf(
+                        realmDictionaryEntryOf(dataSet[0].first, dataSet[0].second),
+                        realmDictionaryEntryOf(dataSet[0].first, dataSet[1].second)
+                    )
+                    assertTrue(entries.removeAll(entriesToRemove))
+                    assertEquals(dictionary.size, entries.size)
+
+                    // Check we get false if we don't remove anything
+                    assertFalse(entries.removeAll(entriesToRemove))
+                }
+            }
+
+            assertContainerAndCleanup { container ->
+                val entries = typeSafetyManager.getCollection(container)
+                    .entries
+
+                val alreadyDeleted = listOf(
+                    realmDictionaryEntryOf(dataSet[0].first, dataSet[0].second)
+                )
+                assertFalse(entries.removeAll(alreadyDeleted))
+
+                val notPresent = listOf(
+                    realmDictionaryEntryOf(dataSet[0].first, dataSet[0].second)
+                )
+                assertFalse(entries.removeAll(notPresent))
+
+                val shouldThrow = listOf(
+                    realmDictionaryEntryOf(dataSet[1].first, dataSet[1].second),
+                )
+                // TODO revisit exception assertion once unified error handling is merged
+                assertFailsWith<RealmException> {
+                    entries.removeAll(shouldThrow)
+                }
+            }
+        }
+    }
+
+    override fun entry_equals() {
+        // Ignore ByteArray and RealmObject: structural equality cannot be assessed for these types
```

Comment:
It doesn't look like these are being tested elsewhere? But I assume it is coming in a later PR.

Reply:
Correct. Equality for byte arrays is fixed in #1254